### ISSUE TITLE
[Dockerfile] #2231 is breaking FIN build stage

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#2233]
@pfultz2 Ths update in #2231 seems to have broken the following stage in docker build:

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/d6b7c41b409d6151c89f37ac0344e7171e6ad1eb/Dockerfile#L100-L104

It can be reproduced by
```
sudo docker build --network=host --build-arg BUILDKIT_INLINE_CACHE=1 --build-arg PREFIX=/opt/rocm --build-arg GPU_ARCH='gfx900;gfx906;gfx908;gfx90a;gfx90a:xnack-;gfx1030' --build-arg MIOTENSILE_VER='default' --build-arg USE_TARGETID='OFF' --build-arg USE_MLIR='ON' --build-arg USE_FIN='ON' .
```

and the error messages are:
[Error](http://micimaster.amd.com/blue/rest/organizations/jenkins/pipelines/MLLibs/pipelines/MIOpen/branches/update_ci_551/runs/4/nodes/6/log/?start=0)

```
> [22/28] RUN if [ "ON" = "ON" ]; then         rbuild prepare -s fin -d /opt/rocm -DAMDGPU_TARGETS=gfx900;gfx906;gfx908;gfx90a;gfx90a:xnack-;gfx1030 -DCMAKE_CXX_COMPILER_LAUNCHER="";     else         rbuild prepare -s develop -d /opt/rocm -DAMDGPU_TARGETS=gfx900;gfx906;gfx908;gfx90a;gfx90a:xnack-;gfx1030 -DCMAKE_CXX_COMPILER_LAUNCHER="";     fi:                                                          
#0 0.157 Traceback (most recent call last):                                                                                                                                                                
#0 0.157   File "/usr/local/bin/rbuild", line 8, in <module>
#0 0.157     sys.exit(cli())
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 829, in __call__
#0 0.157     return self.main(*args, **kwargs)
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 782, in main
#0 0.157     rv = self.invoke(ctx)
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1259, in invoke
#0 0.157     return _process_result(sub_ctx.command.invoke(sub_ctx))
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1066, in invoke
#0 0.157     return ctx.invoke(self.callback, **ctx.params)
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 610, in invoke
#0 0.157     return callback(*args, **kwargs)
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/rbuild/cli.py", line 298, in w
#0 0.157     f(make_builder, *args, **kwargs)
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/rbuild/cli.py", line 310, in prepare
#0 0.157     b = builder()
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/rbuild/cli.py", line 297, in make_builder
#0 0.157     return Builder(session=s, deps_dir=deps_dir, source_dir=source_dir, build_dir=build_dir, toolchain=toolchain, cxx=cxx, cc=cc, define=define)
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/rbuild/cli.py", line 192, in __init__
#0 0.157     session_options = get_session_options(session or 'default', defaults=default_options)
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/rbuild/cli.py", line 157, in get_session_options
#0 0.157     parser = get_config_parser(file, defaults=defaults)
#0 0.157   File "/usr/local/lib/python3.8/dist-packages/rbuild/cli.py", line 133, in get_config_parser
#0 0.157     parser.read([f])
#0 0.157   File "/usr/lib/python3.8/configparser.py", line 697, in read
#0 0.157     self._read(fp, filename)
#0 0.157   File "/usr/lib/python3.8/configparser.py", line 1082, in _read
#0 0.157     raise MissingSectionHeaderError(fpname, lineno, line)
#0 0.157 configparser.MissingSectionHeaderError: File contains no section headers.
#0 0.157 file: '/rbuild.ini', line: 1
#0 0.157 'recipe_hash=7c09d33ac418c31cd5fe1f6e3c2605061dd367c6\n'
```

It looks like this change may have caused this issue:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/d6b7c41b409d6151c89f37ac0344e7171e6ad1eb/rbuild.ini#L1

**Comment by junliume** [comment]
A temporal workaround is to remove the line in rbuild.ini and point to the recipe_hash explicitly.
However, hope there are more elegant solutions available.

**Comment by junliume** [comment]
fixed by #2235
